integration/generic-tests/kexec_test.go

```diff
@@ -144,3 +146,51 @@ func TestMountKexecLoadCustomDTB(t *testing.T) {
 		t.Fatal(err)
 	}
 }
+
+func TestKexecLinuxImageCfgFile(t *testing.T) {
+	// TODO: support arm
+	if vmtest.TestArch() != "amd64" && vmtest.TestArch() != "arm64" {
+		t.Skipf("Test not supported on %s", vmtest.TestArch())
+	}
+
+	testdataDir, err := os.MkdirTemp(t.TempDir(), "test-kexec-linux-image-cfg")
+	if err != nil {
+		t.Fatalf("Failed to create test dir: %v", err)
+	}
+	cfgF, err := os.OpenFile(filepath.Join(testdataDir, "linux_image_cfg.json"), os.O_TRUNC|os.O_CREATE|os.O_WRONLY, 0777)
+	if err != nil {
+		t.Fatalf("Failed to create linux image cfg file: %v", err)
+	}
+	cfg := []byte("{ \"InitrdPath\": \"/testdata/initramfs.cpio\", \"KernelPath\": \"/testdata/kernel\", \"Cmdline\": \"/proc/cmdline\", \"Name\": \"testloadconfig\" }")
+	nw, err := cfgF.Write(cfg)
+	if err != nil {
+		t.Fatalf("Failed to write test cfg: %v", err)
+	}
+	if len(cfg) != nw {
+		t.Fatalf("Written %d bytes, want bytes %d", nw, len(cfg))
+	}
+	if err := cfgF.Sync(); err != nil {
+		t.Fatalf("Failed to flush test cfg file: %v", err)
+	}
+	if err := cfgF.Close(); err != nil {
+		t.Fatalf("Failed to close test cfg file: %v", err)
+	}
```

Comment:
simplify:

cfg := []byte("{ \"InitrdPath\": \"/testdata/initramfs.cpio\", \"KernelPath\": \"/testdata/kernel\", \"Cmdline\": \"/proc/cmdline\", \"Name\": \"testloadconfig\" }")
if err := os.WriteFile(filepath.Join(testdataDir, "linux_image_cfg.json"), cfg, 0777); err != nil {

Reply:
Done! 

WriteFile does not do Sync before Close -- but  I don't see practical issue with test file in tmpfs.